easybuild/tools/options.py

```diff
@@ -724,6 +729,29 @@ def postprocess(self):
             self.show_config()
             cleanup_and_exit(self.tmpdir)
 
+    def _postprocess_optarch(self):
+        """Postprocess --optarch option."""
+        optarch_parts = self.options.optarch.split(OPTARCH_SEP)
+        
+        # we expect to find a ':' in every entry in optarch, in case optarch is specified on a per-compiler basis
+        if len(optarch_parts) > 1 and not all(OPTARCH_MAP_CHAR in p for p in optarch_parts):
```

Comment:
@damianam I gave this some more thought, and I actually think we need a more complex check to *really* validate this... We should check that there *exactly* one `OPTARCH_MAP_CHAR` in each part, now we may have multiple, which will screw things up later where we use `.split(OPTARCH_MAP_CHAR)`...